### PR TITLE
Add UDP broadcast mode

### PR DESCRIPTION
**Pull request by dvornikov-aa**
UDP endpoint updates target IP address from the incoming MAVLink message. This PR adds a new broadcast mode that has a fixed target IP.

### CODE REVIEW

**Comment by lucasdemarchi** [comment]
The name seems a bit confusing since it may be thought of as the broadcast address. Maybe "Fixed" would be better? 

What's the use case of having it fixed?

File: src/mavlink-router/endpoint.cpp [hunk 1]
```diff
@@ -731,8 +735,12 @@ int UdpEndpoint::open(const char *ip, unsigned long port, bool to_bind)
 ssize_t UdpEndpoint::_read_msg(uint8_t *buf, size_t len)
 {
     socklen_t addrlen = sizeof(sockaddr);
-    ssize_t r = ::recvfrom(fd, buf, len, 0,
-                           (struct sockaddr *)&sockaddr, &addrlen);
+    ssize_t r;
+    if (_mode == UdpEndpoint::Broadcast)
+        r = ::recvfrom(fd, buf, len, 0, NULL, NULL);
+    else
+        r = ::recvfrom(fd, buf, len, 0,
```
Comment (lucasdemarchi) [inline src/mavlink-router/endpoint.cpp]: This change is not needed. You can just ignore the values later

**Comment by dvornikov-aa** [comment]
I thought "Broadcast" is more obvious (at least you see the purpose of this mode). Actually, I like "Fixed" more. For example, https://github.com/MonashUAS/cmavnode project names this mode "bcastlock", https://github.com/mavlink/mavros - "permanent broadcast" (https://github.com/mavlink/mavros/pull/1043).

This mode is useful when you want multiple vehicles to use one endpoint (multiple GCS with different MAVLink IDs, direct vehicle communication without routing).

File: src/mavlink-router/endpoint.cpp (hunk 1 above)
Comment (dvornikov-aa) [reply src/mavlink-router/endpoint.cpp]: sockaddr contains a target IP address. We rewrite it on every incoming packet (recvfrom). To ignore this change we need to add an additional sockaddr structure in the endpoint class.

https://github.com/intel/mavlink-router/blob/da8607430c9e336e826d576c15f555c5f0fe033c/src/mavlink-router/endpoint.cpp#L761

**Review by JonasVautherin — COMMENTED** [review]
I agree on the fact that setting a broadcast address in "Normal" mode will not always work as one would expect, because the messages are sent to the address of the last received message.

So if you set the target address as `192.168.1.255/24`, and you have two clients A (`192.168.1.10/24`) and B (`192.168.1.11/24`), both A and B will receive the first few messages. Whenever mavlink-router receives a message from A, it will replace the target address with `192.168.1.10/24` and B will stop receiving messages. Similarly when it receives a message from B. Because probably both A and B will continuously send messages, the target address will flip between both and the behavior will be unexpected.

This said, I don't understand the need for a "Broadcast" mode. Why not always using `recv()` instead of `recvfrom()` and never change the target address? This way the broadcast may just work, right? Or am I missing something?

**Comment by jbeyerstedt** [comment]
I'll close this for now because of missing activity. Please continue the discussion about the functionality itself in #350 and see the notes on another implementation attempt there.
(Or just add a comment here, if closing was a mistake.)

**Pull request closed without merging**